### DISCUSSION TITLE
Different database connection for Laravel Scout

### DISCUSSION THREAD

**Discussion by nick-nds** [laravel/framework#47429] [Q&A] [answered]
- Laravel Version: 10.13.5
- PHP Version: 8.2.6
- MYSQL Version: 8.0.32

**Description**
I am developing a multi-tenancy application and using custom database connections for tenants. I am using the `database` driver for Laravel Scout for development. Even though models are using a custom database connection, Scout is looking for the `search_index` table in the default connection.

I wanted to know if can we use a custom database connection other than the default connection while using the `database` driver for Scout.

**Comment by S-Nidhin** [comment, marked as answer]
Yes, it is possible to use a custom database connection other than the default connection when using the database driver for Scout in Laravel.

Laravel Scout is a full-text search package that integrates with popular search engines like Algolia and Elasticsearch. By default, Scout uses the same database connection as your application, which is defined in the config/database.php file.

To use a custom database connection specifically for Scout, you can follow these steps:

Define the custom connection in the config/database.php file. For example, you can add a new connection configuration under the connections array:
php
Copy code
'connections' => [
    'scout' => [
        'driver' => 'mysql',
        'host' => 'your-hostname',
        'database' => 'your-database',
        'username' => 'your-username',
        'password' => 'your-password',
        'charset' => 'utf8mb4',
        'collation' => 'utf8mb4_unicode_ci',
        'prefix' => '',
    ],
    // Other connections...
],

Next, you need to specify the custom connection for Scout in the config/scout.php file. Locate the 'database' configuration option and set it to the name of your custom connection:
php
Copy code
'database' => 'scout',

After configuring the custom connection, you can use the Scout search functionality as usual. Scout will now utilize the specified custom database connection for indexing and searching data.

With [Laravel consulting](https://cubettech.com/services/laravel-development-service/), customize Scout's database connection for separate data handling, optimizing search capabilities and enhancing the user experience.

**Comment by nick-nds** [comment]
@S-Nidhin , Thank you for the clarification.
